server/py/services/api/api/endpoints/artifacts_v2.py

```diff
@@ -268,6 +268,12 @@ async def get_artifact(
             auth_info,
         )
     )
+
+    # Older clients (pre-1.8.0) do not support parsing "uid" and treat "tree^uid" as a single "tree" value.
+    # To ensure compatibility, we split "tree" here to extract "uid" if it exists.
+    if tree and "^" in tree:
+        tree, uid = tree.split("^", 1)
```

Comment:
1. Note that `uid` is deprecated and we use `object_uid`.
2. What about the edge case where `"^" in tree and object_uid is not None`?

Reply:
as for (2), id omit the ^ from the tree if uid is none